x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -689,23 +689,37 @@ public Expression visitStringLiteral(StringLiteralContext ctx) {
 
     @Override
     public Literal visitDecimalLiteral(DecimalLiteralContext ctx) {
-        String string = (hasMinusFromParent(ctx) ? "-" : "") + ctx.getText();
+        String string = ctx.getText();
```

Comment:
Since this code is repeated in the `visitIntegerLiteral`, maybe it can be encapsulated in the `minusAwareSource` or in a new method that wraps it.